apps/smeargle/src/components/Filter/FilterDetailed/useComboBoxInput.ts

```diff
@@ -0,0 +1,106 @@
+import { createSignal, onMount } from 'solid-js';
+import { FilterComboBoxProps, FilterName, FilterOption } from '../types';
+import { useFilterContext } from '../FilterProvider';
+import { TeaserRequestParams } from 'tonemato-types';
+import { useFilterDetailed } from './useFilterDetailed';
+
+export function useComboBoxInput(
+  props: FilterComboBoxProps<TeaserRequestParams[FilterName]>
+) {
+  const [inputValue, setInputValue] = createSignal(
+    props.options ? props.options[props.value] : props.value
+  );
+  const [filterState, { updateValue }] = useFilterContext();
+
+  const { focusNextSection } = useFilterDetailed();
+
+  onMount(() => {
+    if (props.value) {
+      updateValue(props.name, props.value);
+    }
+  });
+
+  const isInputInOptions = () => {
+    if (props.isDropdown) return false;
+    if (!props.options) return false;
+    return (
+      Object.entries(props.options).filter(([, text]) => text === inputValue())
```

Comment:
Sometimes we have `[_, otherVar]` and sometimes `[, otherVar]`. Also in `useFilterDetailed.ts` but i can't add a suggestion in this, lel.
```suggestion
      Object.entries(props.options).filter(([_, text]) => text === inputValue())
```

Reply:
jup thats a default in other languages but ts is mad if I reuse _ as a variable... just gonna remove them